wire-runtime/src/test/proto-kotlin/com/squareup/wire/protos/kotlin/person/Person.kt.java.interop

```diff
@@ -0,0 +1,225 @@
+        fun phone(phone: List<PhoneNumber>): Builder {
+            Internal.checkElementsNotNull(phone)
+            this.phone = phone
+            return this
+        }
+
+        override fun build(): Person = Person(
+            name = name ?: throw Internal.missingRequiredFields(name, "name"),
+            id = id ?: throw Internal.missingRequiredFields(id, "id"),
+            email = email,
+            phone = phone,
+            unknownFields = super.buildUnknownFields()
+        )
+    }
+
+    companion object {
+        @JvmField
+        val ADAPTER: ProtoAdapter<Person> =
+                object : ProtoAdapter<Person>(FieldEncoding.LENGTH_DELIMITED, Person::class.java) {
+            override fun encodedSize(value: Person): Int = ProtoAdapter.STRING.encodedSizeWithTag(1, value.name) +
+                ProtoAdapter.INT32.encodedSizeWithTag(2, value.id) +
+                ProtoAdapter.STRING.encodedSizeWithTag(3, value.email) +
+                PhoneNumber.ADAPTER.asRepeated().encodedSizeWithTag(4, value.phone) +
+                value.unknownFields.size()
+
+            override fun encode(writer: ProtoWriter, value: Person) {
+                ProtoAdapter.STRING.encodeWithTag(writer, 1, value.name)
+                ProtoAdapter.INT32.encodeWithTag(writer, 2, value.id)
+                ProtoAdapter.STRING.encodeWithTag(writer, 3, value.email)
+                PhoneNumber.ADAPTER.asRepeated().encodeWithTag(writer, 4, value.phone)
+                writer.writeBytes(value.unknownFields)
+            }
+
+            override fun decode(reader: ProtoReader): Person {
+                var name: String? = null
+                var id: Int? = null
+                var email: String? = null
+                var phone = mutableListOf<PhoneNumber>()
+                val unknownFields = reader.forEachTag { tag ->
+                    when (tag) {
+                        1 -> name = ProtoAdapter.STRING.decode(reader)
+                        2 -> id = ProtoAdapter.INT32.decode(reader)
+                        3 -> email = ProtoAdapter.STRING.decode(reader)
+                        4 -> phone.add(PhoneNumber.ADAPTER.decode(reader))
+                        else -> TagHandler.UNKNOWN_TAG
+                    }
+                }
+                return Person(
+                    name = name ?: throw Internal.missingRequiredFields(name, "name"),
+                    id = id ?: throw Internal.missingRequiredFields(id, "id"),
+                    email = email,
+                    phone = phone,
+                    unknownFields = unknownFields
+                )
+            }
+        }
+    }
+
+    enum class PhoneType(private val value: Int) : WireEnum {
+        MOBILE(0),
+
+        HOME(1),
+
+        /**
+         * Could be phone or fax. */
+        WORK(2);
+
+        override fun getValue(): Int = value
+
+        object ADAPTER : EnumAdapter<PhoneType>(PhoneType::class.java) {
```

Comment:
Btw is it Java-compatible to have `ADAPTER` as an `object` inside a `WireEnum`? Or should it be a `@JvmField` inside a `companion object`?